api/dataset_api.go

```diff
@@ -0,0 +1,133 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/ONSdigital/go-ns/log"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+// DatasetAPI aggregates a client and URL and other common data for accessing the API
+type DatasetAPI struct {
+	Client    *http.Client
+	URL       string
+	AuthToken string
+}
+
+// NewDatasetAPI creates an DatasetAPI object
+func NewDatasetAPI(client *http.Client, url string, authToken string) *DatasetAPI {
+	return &DatasetAPI{
+		Client:    client,
+		URL:       url,
+		AuthToken: authToken,
+	}
+}
+
+// InstanceResults wraps instances objects for pagination
+type InstanceResults struct {
+	Items []Instance `json:"items"`
+}
+
+// Instance comes in results from the Dataset API
+type Instance struct {
+	InstanceID                string `json:"instance_id"`
+	NumberOfObservations      int64  `json:"total_observations"`
+	TotalInsertedObservations int64  `json:"total_inserted_observations,omitempty"`
+	Job                       IDLink `json:"job"`
```

Comment:
The json in the dataset-api has changes to 
```
{
	"links": {
		"job": {
			"id": "123-456",
			"href": "http://localhost:2200/jobs/123-456"
		}
	}
}
```

Reply:
I have updated to use the latest model 👍